src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelBindingContext.cs

```diff
@@ -27,60 +22,75 @@ public ModelBindingContext()
         /// <returns>A new instance of <see cref="ModelBindingContext"/>.</returns>
-        public static ModelBindingContext GetModelBindingContext(
+        public static ModelBindingContext CreateBindingContext(
+            [NotNull] OperationBindingContext operationBindingContext,
+            [NotNull] ModelStateDictionary modelState,
             [NotNull] ModelMetadata metadata,
             BindingInfo bindingInfo,
-            string modelName)
+            [NotNull] string modelName)
         {
             var binderModelName = bindingInfo?.BinderModelName ?? metadata.BinderModelName;
             var propertyPredicateProvider =
                 bindingInfo?.PropertyBindingPredicateProvider ?? metadata.PropertyBindingPredicateProvider;
-            return new ModelBindingContext
+
+            return new ModelBindingContext()
             {
-                ModelMetadata = metadata,
+                BinderModelName = binderModelName,
                 BindingSource = bindingInfo?.BindingSource ?? metadata.BindingSource,
-                PropertyFilter = propertyPredicateProvider?.PropertyFilter,
                 BinderType = bindingInfo?.BinderType ?? metadata.BinderType,
-                BinderModelName = binderModelName,
-                ModelName = binderModelName ?? metadata.PropertyName ?? modelName,
+                PropertyFilter = propertyPredicateProvider?.PropertyFilter,
+
+                // We only support fallback to empty prefix in cases where the model name is inferred from
+                // the parameter or property being bound.
                 FallbackToEmptyPrefix = binderModelName == null,
+
+                // Because this is the top-level context, FieldName and ModelName should be the same.
+                FieldName = binderModelName ?? modelName,
+                ModelName = binderModelName ?? modelName,
+
+                IsTopLevelObject = true,
+                ModelMetadata = metadata,
+                ModelState = modelState,
+                OperationBindingContext = operationBindingContext,
+                ValueProvider = operationBindingContext.ValueProvider,
+            };
+        }
+
+        public static ModelBindingContext CreateChildBindingContext(
+            [NotNull] ModelBindingContext parent,
+            [NotNull] ModelMetadata modelMetadata,
+            [NotNull] string fieldName,
+            [NotNull] string modelName,
+            object model)
+        {
+            return new ModelBindingContext()
+            {
+                IsFirstChanceBinding = parent.IsFirstChanceBinding,
+                ModelState = parent.ModelState,
+                OperationBindingContext = parent.OperationBindingContext,
+                ValueProvider = parent.ValueProvider,
+
+                // These are never true for a child context.
+                FallbackToEmptyPrefix = false,
+                IsTopLevelObject = false,
```

Comment:
I thought it added clarity. Honestly I wasn't really thinking about it, just making sure everything was initialized.